server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramAggregationBuilder.java

```diff
@@ -417,9 +417,28 @@ protected ValuesSourceAggregatorFactory innerBuild(
         AggregatorFactory parent,
         AggregatorFactories.Builder subFactoriesBuilder
     ) throws IOException {
-        DateHistogramAggregationSupplier aggregatorSupplier = context.getValuesSourceRegistry().getAggregator(REGISTRY_KEY, config);
+        final DateIntervalWrapper.IntervalTypeEnum dateHistogramIntervalType = dateHistogramInterval.getIntervalType();
+        context.getIndexSettings()
+            .getMode()
+            .validateCalendarIntervalType(
+                dateHistogramIntervalType,
+                config.getDescription()
```

Comment:
I don't love this pattern of passing the message in and throwing from the validator.  Why not just have the validator return a boolean and throw from here if we want to?  I don't feel terribly strongly about it though.

Reply:
:rofl: I advocated for having the method do the throwing. I think it's *generally* a cleaner way to do it, especially because the method has to return a very "related" type of exception. So! Maybe it won't matter. This is the thing we're talking about maybe not applying correctly anyway - I think we only want to throw if we're operating on a rolled up index. Not a timeseries one. And *that* will work differently anyway.

Reply:
I think I did something like this before and I changed it. I find it weird to return a boolean from a validation method.
Should the method return true or false if validation succeeds?

Reply:
Maybe I'm splitting hairs, but if the validation method is going to throw, it should own building the message.  That's all I'm saying.  Having the message in one place and the decision to throw in a different place is weird to me.